base/src/main/java/org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -1796,42 +1810,29 @@ public int check(SSLFDProxy fd) {
                         // Since we're a server validating the client's
                         // chain (and they didn't provide one), we should
                         // ignore it instead of forcing the problem.
-                        debug("No client certificate chain and client cert not needed.");
+                        logger.debug("No client certificate chain and client cert not needed.");
                         return 0;
                     }
                 }
 
                 for (X509TrustManager tm : trust_managers) {
-                    // X509ExtendedTrustManager lets the TM access the
-                    // SSLEngine while validating certificates. Otherwise,
-                    // the X509TrustManager doesn't have access to that
-                    // parameter. Facilitate it if possible.
-                    if (tm instanceof X509ExtendedTrustManager) {
-                        X509ExtendedTrustManager etm = (X509ExtendedTrustManager) tm;
-                        if (as_server) {
-                            etm.checkClientTrusted(chain, authType, JSSEngineReferenceImpl.this);
-                        } else {
-                            etm.checkServerTrusted(chain, authType, JSSEngineReferenceImpl.this);
-                        }
+                    if (as_server) {
+                        tm.checkClientTrusted(chain, authType);
                     } else {
-                        if (as_server) {
-                            tm.checkClientTrusted(chain, authType);
-                        } else {
-                            tm.checkServerTrusted(chain, authType);
-                        }
+                        tm.checkServerTrusted(chain, authType);
                     }
                 }
             } catch (Exception excpt) {
-                return assignException(excpt, chain);
+                return logException(excpt, chain);
             }
 
             return 0;
         }
 
-        private int assignException(Exception excpt, PK11Cert[] chain) {
+        private int logException(Exception excpt, PK11Cert[] chain) {
             int nss_code = Cert.MatchExceptionToNSSError(excpt);
 
-            if (seen_exception) {
+            if (logger.isDebugEnabled() == false) {
```

Comment:
I think this can be written as:
```
if (!logger.isDebugEnabled()) {
```

Reply:
Yep.  It's a habit I formed a long time ago.  I'll change it.

Note.  Sonar doesn't like that this method returns nss_code in 2 places.  It's saying it's a "Blocker".
https://sonarcloud.io/project/issues?pullRequest=928&issues=AYWX5DBvciWndDK-ESS5&open=AYWX5DBvciWndDK-ESS5&id=dogtagpki_jss

I could change it to:
```
if (logger.isDebugEnabled()) {
  String msg = ...
  logger.debug(msg, excpt);
}
return nss_code;
```

I wouldn't ordinarily write code like that though, and the code wasn't originally ordered like that either.

Reply:
Sonar isn't marking it as a blocker any more...